src/game/bank/bank.cpp

```diff
@@ -156,10 +156,11 @@ bool Bank::deposit(const std::shared_ptr<Bank> &destination) {
 	if (!bankable) {
 		return false;
 	}
-	if (bankable->getPlayer() == nullptr) {
+	auto player = bankable->getPlayer();
+	if (!player) {
 		return false;
 	}
-	auto amount = bankable->getPlayer()->getMoney();
+	auto amount = player->getMoney();
```

Comment:
This change is indeed beneficial, but it doesn't necessarily resolve the issue, as a nullptr check was already in place, so there would be no pointer dereference. However, it's beneficial to centralize it into a single call, thereby avoiding two consecutive calls to "getPlayer," which allocates the pointer, reference counter, etc.